chandra_aca/dark_subtract.py

```diff
@@ -0,0 +1,431 @@
+from functools import lru_cache
+
+import numba
+import numpy as np
+import requests.exceptions
+from astropy.table import Table, vstack
+from cheta import fetch_sci
+from cxotime import CxoTime, CxoTimeLike
+from mica.archive import aca_l0
+from mica.archive.aca_dark import get_dark_cal_props
+from ska_helpers import retry
+
+from chandra_aca import maude_decom
+from chandra_aca.dark_model import dark_temp_scale_img
+
+
+def get_dark_data(time: CxoTimeLike):
+    """
+    Retrieve the dark calibration image and temperature for a given time.
+
+    This is a light wrapper around mica.archive.aca_dark.get_dark_cal_props.
+
+    Parameters
+    ----------
+    time : CxoTimeLike
+        Reference time for the dark calibration image.
+
+    Returns
+    -------
+    dc_img : np.array
+        Dark calibration image.
+    dc_tccd : float
+        Dark calibration CCD temperature.
+    """
+    dc = get_dark_cal_props(time, "nearest", include_image=True, aca_image=False)
+    dc_img = dc["image"]
+    if "t_ccd" in dc:
+        dc_tccd = dc["t_ccd"]
+    elif "ccd_temp" in dc:
+        dc_tccd = dc["ccd_temp"]
+    else:
+        raise ValueError("Dark cal must have t_ccd or ccd_temp")
+    return dc_img, dc_tccd
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+def get_tccd_data(start: CxoTimeLike, stop: CxoTimeLike, source="maude"):
+    """
+    Get the CCD temperature for a given time range.
+
+    This is a light wrapper around cheta.fetch_sci.Msid("aacccdpt", start, stop).
+
+    Parameters
+    ----------
+    start : CxoTimeLike
+        Start time of the time range.
+    stop : CxoTimeLike
+        Stop time of the time range.
+    source : str, optional
+        Source of the CCD temperature data. If 'maude', override the fetch_sci data source
+        to 'maude allow_subset=False'. Else, use the cheta default.
+
+    Returns
+    -------
+    t_ccd : np.array
+        CCD temperature values.
+    t_ccd_times : np.array
+        Times corresponding to the CCD temperature values.
+    """
+    if source == "maude":
+        # Override the data_source to be explicit about maude source.
+        with fetch_sci.data_source("maude allow_subset=False"):
+            dat = fetch_sci.Msid("aacccdpt", start, stop)
+    else:
+        dat = fetch_sci.Msid("aacccdpt", start, stop)
+    t_ccd = dat.vals
+    t_ccd_times = dat.times
+    return t_ccd, t_ccd_times
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+def get_dcsub_aca_images(
```

Comment:
Let's make this a more private method and then have the public method just have start, stop, source options.